File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n=nums.size();

        unordered_map<int,int>map;

        for(int i=0;i<nums.size();i++){
            map[nums[i]]++;
        }

        int ans=0;

        for(int i=0;i<n;i++){
            if(map.count(nums[i]-1))continue;
            int cnt=0;
            int val=nums[i];
            while(map.count(val)){
                cnt++;
                val++;
            }
            ans=max(ans,cnt);
            if(cnt>nums.size()/2)break;
        }

        return ans;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int ans=0;
        int n=nums.length;

        Map<Integer,Integer> map = new HashMap<>();

        for(int i=0;i<n;i++){
            map.put(nums[i],1);
        }

        for(int i=0;i<n;i++){
            if(map.containsKey(nums[i]-1))continue;
            int currNum=nums[i];
            int currcnt=0;

            while(map.containsKey(currNum)){
                currNum++;
                currcnt++;
            }
            ans=Math.max(ans,currcnt);
            if(currcnt>n/2)break;
        }

        return ans;
    }
}